### PR TITLE
Feature/organization permissions

### PR DESCRIPTION
**Pull request by blootsvoets**
Description:
- Add organization permissions.
- Add organization_admin role.
- Refactor permissions to make use of relevant tech (enum, enummap).

#### Checklist:
- [x] The [Main workflow](https://github.com/RADAR-base/ManagementPortal/actions/workflows/main.yml) has succeeded
- [x] The [Gatling tests](https://github.com/RADAR-base/ManagementPortal#other-tests) have passed
- [x] I have logged into the portal running locally with default admin credentials
- [x] I have updated the README files if this change requires documentation update
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: radar-auth/build.gradle [hunk 1]
```diff
@@ -14,8 +14,8 @@ ext {
     junitVersion ='5.6.3'
 }
 
-sourceCompatibility = JavaVersion.VERSION_1_8
-targetCompatibility = JavaVersion.VERSION_1_8
+sourceCompatibility = JavaVersion.VERSION_11
```
Comment (aykl) [inline radar-auth/build.gradle:17]: Wouldn't it make sense to use the same version across subprojects?

Comment (blootsvoets) [reply radar-auth/build.gradle:17]: The reason for the difference is that the libraries need some more compatibility, and application code can use the latest technologies.

File: src/main/java/org/radarbase/management/service/UserService.java [hunk 2]
```diff
@@ -198,8 +190,9 @@ public User createUser(UserDTO userDto) {
         Set<Role> roles = new HashSet<>();
         for (RoleDTO roleDto : userDto.getRoles()) {
             Optional<Role> role = roleRepository.findOneByProjectIdAndAuthorityName(
-                    roleDto.getProjectId(), roleDto.getAuthorityName());
-            if (!role.isPresent() || role.get().getId() == null) {
+                    roleDto.getProjectId(), roleDto.getAuthorityName())
+                    .filter(r -> r.getId() != null);
```
[the remark this answers is not in the document]
Comment (blootsvoets) [reply src/main/java/org/radarbase/management/service/UserService.java:194]: It duplicates the code above it -- I'm not sure whether it's ever null...

**Review by aykl — APPROVED** [review]
LGTM.

File: src/test/java/org/radarbase/management/web/rest/OrganizationResourceIntTest.java [hunk 3]
```diff
@@ -7,6 +7,7 @@
 import org.mockito.MockitoAnnotations;
 import org.radarbase.auth.authentication.OAuthHelper;
 // import org.radarbase.auth.token.RadarToken;
```
Comment (aykl) [inline src/test/java/org/radarbase/management/web/rest/OrganizationResourceIntTest.java:9]:
```suggestion
```

**Pull request merged**